src/openrct2/drawing/Drawing.h

```diff
@@ -601,7 +601,7 @@ void gfx_draw_pickedup_peep(rct_drawpixelinfo* dpi);
 void gfx_draw_line(rct_drawpixelinfo* dpi, int32_t x1, int32_t y1, int32_t x2, int32_t y2, int32_t colour);
 void gfx_draw_line_software(rct_drawpixelinfo* dpi, int32_t x1, int32_t y1, int32_t x2, int32_t y2, int32_t colour);
 void gfx_draw_dashed_line(
-    rct_drawpixelinfo* dpi, int32_t x1, int32_t y1, int32_t x2, int32_t y2, int32_t dashedLineSegmentLength, int32_t colour);
+    rct_drawpixelinfo* dpi, const ScreenLine& screenLine, const int32_t dashedLineSegmentLength, const int32_t color)
```

Comment:
Go to line 21 of this file and add this afterwards to fix compilation:
```
    struct ScreenLine;
```

This is called a forward declaration, you provide a type to the compiler, even if incomplete, so you can use reference and pointers of it in the header without actually including its definition. It's used to reduce the number of includes in header files and make compilations generally faster when it's altered.

Reply:
Is this something I missed learning c? Or is this a feature of cpp?
And should then ScreenCoordsXY be forward declared twice as in line 19 and 21 of Drawing.h?

Reply:
Ahh so I just learned it under another name and I learned to use is in only specific ways. Great that I can expand on that.

Reply:
Nice :)